tests/e2e/p/workflow.go

```diff
@@ -25,177 +24,163 @@ import (
+				Wght:   minValStake,
+			},
+			Subnet: constants.PrimaryNetworkID,
+		}
+
+		tc.By("issuing an AddPermissionlessValidatorTx", func() {
 			sk, err := bls.NewSecretKey()
 			require.NoError(err)
 			pop := signer.NewProofOfPossession(sk)
 
-			tc.By("issue add validator tx", func() {
-				_, err := pWallet.IssueAddPermissionlessValidatorTx(
-					vdr,
-					pop,
-					avaxAssetID,
-					rewardOwner,
-					rewardOwner,
-					shares,
-					tc.WithDefaultContext(),
-				)
-				require.NoError(err)
-			})
-
-			tc.By("issue add delegator tx", func() {
-				_, err := pWallet.IssueAddPermissionlessDelegatorTx(
-					vdr,
-					avaxAssetID,
-					rewardOwner,
-					tc.WithDefaultContext(),
-				)
-				require.NoError(err)
-			})
-
-			// retrieve initial balances
-			pBalances, err := pWallet.Builder().GetBalance()
+			_, err = pWallet.IssueAddPermissionlessValidatorTx(
+				vdr,
+				pop,
+				avaxAssetID,
+				rewardOwner,
+				rewardOwner,
+				delegationFeeShares,
+				tc.WithDefaultContext(),
+			)
 			require.NoError(err)
-			pStartBalance := pBalances[avaxAssetID]
-			tc.Outf("{{blue}} P-chain balance before P->X export: %d {{/}}\n", pStartBalance)
+		})
 
-			xBalances, err := xWallet.Builder().GetFTBalance()
+		tc.By("issuing an AddPermissionlessDelegatorTx", func() {
+			_, err := pWallet.IssueAddPermissionlessDelegatorTx(
+				vdr,
+				avaxAssetID,
+				rewardOwner,
+				tc.WithDefaultContext(),
+			)
 			require.NoError(err)
-			xStartBalance := xBalances[avaxAssetID]
-			tc.Outf("{{blue}} X-chain balance before P->X export: %d {{/}}\n", xStartBalance)
+		})
 
-			outputOwner := secp256k1fx.OutputOwners{
-				Threshold: 1,
-				Addrs: []ids.ShortID{
-					xTargetAddr,
-				},
-			}
-			output := &secp256k1fx.TransferOutput{
-				Amt:          toTransfer,
-				OutputOwners: outputOwner,
-			}
+		tc.By("issuing an ExportTx on the P-chain", func() {
+			balances, err := pBuilder.GetBalance()
+			require.NoError(err)
+
+			initialAVAXBalance := balances[avaxAssetID]
+			tc.Outf("{{blue}} P-chain balance before P->X export: %d {{/}}\n", initialAVAXBalance)
 
-			tc.By("export avax from P to X chain", func() {
-				_, err := pWallet.IssueExportTx(
-					xContext.BlockchainID,
-					[]*avax.TransferableOutput{
-						{
-							Asset: avax.Asset{
-								ID: avaxAssetID,
-							},
-							Out: output,
+			exportTx, err := pWallet.IssueExportTx(
+				xContext.BlockchainID,
+				[]*avax.TransferableOutput{
+					{
+						Asset: avax.Asset{
+							ID: avaxAssetID,
+						},
+						Out: &secp256k1fx.TransferOutput{
+							Amt:          toTransfer,
+							OutputOwners: transferOwner,
 						},
 					},
-					tc.WithDefaultContext(),
-				)
-				require.NoError(err)
-			})
+				},
+				tc.WithDefaultContext(),
+			)
+			require.NoError(err)
 
-			// check balances post export
-			pBalances, err = pWallet.Builder().GetBalance()
+			exportFee, err := pFeeCalculator.CalculateFee(exportTx.Unsigned)
```

Comment:
This is the functional change for this test. We now use the `pFeeCalculator` rather than referencing the hardcoded static fee.